R/conclass.R

```diff
@@ -408,42 +437,55 @@ Conos <- R6::R6Class("Conos", lock_objects=F,
     },
 
     #' @description  Generate an embedding of a joint graph.
-
+    #'
     #' @param method embedding method. Currently largeVis and UMAP are supported
+    #' @param name optional name of the embedding result (will default to the method name; otherwise, "embedding")
     #' @param M, gamma, alpha, sgd__batched - largeVis parameters (defaults are 1, 1, 0.01, 1e8 respectively).
     #' @param perplexity perplexity passed to largeVis (defaults to NA).
     #' @param seed random seed for the largeVis algorithm. Default: 1.
     #' @param target.dims numer of dimensions for the reduction. Default: 2. Higher dimensions can be used to generate embeddings for subsequent reductions by other methods, such as tSNE
     #' @param n.cores number of cores, overrides class field
     #' @param verbose verbose mode. Default: TRUE.
     #' @param ... additional arguments, passed to UMAP embedding (run ?conos:::embedGraphUmap for more info)
-    embedGraph=function(method='largeVis', M=1, gamma=1, alpha=0.1, perplexity=NA, sgd_batches=1e8, seed=1, verbose=TRUE, target.dims=2, n.cores=self$n.cores, ...) {
+    embedGraph=function(method='largeVis', name=NULL, M=1, gamma=1, alpha=0.1, perplexity=NA, sgd_batches=1e8, seed=1, verbose=TRUE, target.dims=2, n.cores=self$n.cores, ...) {
       supported.methods <- c('largeVis', 'UMAP')
       if(!method %in% supported.methods) { stop(paste0("currently, only the following embeddings are supported: ",paste(supported.methods,collapse=' '))) }
 
+      if (is.null(name)) {
+        name <- method
+        if(is.null(name)) {
+          name <- "embedding"
+        }
+      }
+
       if (method == 'largeVis') {
         wij <- as_adj(self$graph,attr='weight');
         if(!is.na(perplexity)) {
           wij <- conos:::buildWijMatrix(wij,perplexity=perplexity,threads=n.cores)
         }
         coords <- conos:::projectKNNs(wij = wij, dim=target.dims, verbose = verbose,sgd_batches = sgd_batches,gamma=gamma, M=M, seed=seed, alpha=alpha, rho=1, threads=n.cores)
         colnames(coords) <- V(self$graph)$name
-        self$embedding <- t(coords);
+        if (!is.null(self$embedding)){
```

Comment:
This check actually introduces the breaking changes to the code. First, with this, conos itself actually never sets `self$embedding`, so it has to be an user who sets it by hand. So if I'd re-run my old code that uses `con$embedding`, I won't get the same result and it would fail somewhere. Second, if `self$embedding` is not `NULL`, `embedding.result` is not defined and the code below won't work. I think we should remove this `if` and have it as `self$embeddings[[name]] <- self$embedding <- t(coords)`.

Reply:
I think there were some conceptual misunderstandings when I first tried this; let's regroup below.